### PR TITLE
[MNG-5571] The --resume-from hint at the end of a failing build shoul…

### PR DESCRIPTION
**Pull request by MartinKanters**
…d hint the first topologically sorted failed build, instead of the first one chronologically.

### CODE REVIEW

**Comment by rfscholte** [comment]
please rebase

**Comment by MartinKanters** [comment]
> please rebase

@rfscholte done!

File: core-it-suite/src/test/resources/mng-5571-multiple-failed-projects-rf-hint/c/pom.xml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.apache.maven.its.mng5571</groupId>
+        <artifactId>parent</artifactId>
+        <version>0.1-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>c</artifactId>
+    <version>0.1-SNAPSHOT</version>
```
Comment (michael-o) [inline core-it-suite/src/test/resources/mng-5571-multiple-failed-projects-rf-hint/c/pom.xml:31]: You can safely inherit this from parent.

Comment (MartinKanters) [reply core-it-suite/src/test/resources/mng-5571-multiple-failed-projects-rf-hint/c/pom.xml:31]: Nice find! Although I'll have to close this pull request I'm afraid. I'll add a global comment to this PR with the reasoning why.

**Comment by MartinKanters** [comment]
By fixing [MNG-5760](https://issues.apache.org/jira/browse/MNG-5760) we have introduced the `--resume` feature. With this feature the hint at the bottom of the build log will advise to use `-r`. It will only fallback to `-rf :<artifactId>` if the resume.properties file could not be created (because of a filesystem issue). This scenario is very hard, if not impossible, to create in an integration test, so I'm afraid we'll have to close this PR. [The original PR](https://github.com/apache/maven/pull/340) in maven-core is still valid.
If anybody has a suggestion how to fix this, please let me know.

**Pull request closed without merging**